docs/pages/documentation/3-querying-with-graql/dml.md

```diff
@@ -0,0 +1,306 @@
+---
+title: Data Manipulation Language
+keywords: graql, query, data manipulation language
+last_updated: April 2017
+tags: [graql]
+summary: "Data Manipulation Language"
+sidebar: documentation_sidebar
+permalink: /documentation/graql/dml.html
+folder: graql
+---
+
+# Match
+
+`match` [`<pattern> ...`](./query.html#pattern) [`(<modifier> ...)`](#modifier)
+
+A [match](#match) will find all [answers](./query.html#answer) in the knowledge base that _satisfy_ all of the given
+[patterns](#patterns). Any optional [modifiers](#modifier) are applied to the [answers](./query.html#answer) to the
+[match](#match).
+
+The [answers](./query.html#answer) to a [match](#match) can be used with either a [get](#get-query),
+[insert](#insert-query), [delete](#delete-query) or [aggregate](#aggregate-query).
+
+## Pattern
+
+A [pattern](#pattern) is either a [variable pattern](./query.html#variable-pattern), a conjunction or a disjunction:
+
+- Conjunction: `{ <pattern> ... };` - An [answer](./query.html#answer) _satisfies_ a conjunction if it _satisfies_ all
+  [patterns](#pattern) within the conjunction.
+
+- Disjunction: `<pattern> or <pattern>;` - An [answer](./query.html#answer) _satisfies_ a disjunction if it _satisfies_
+  either [pattern](#pattern) within the disjunction.
+
+- An [answer](./query.html#answer) _satisfies_ a [variable pattern](./query.html#variable-pattern) if the concept bound
+  to the subject of the
+  [variable pattern](./query.html#variable-pattern) _satisfies_ all [properties](./query.html#property) in the
+  [variable pattern](./query.html#variable-pattern).
+
+Within the [patterns](#pattern), the following [properties](./query.html#property) are supported:
+
+### isa
+
+`$x isa $A` is _satisfied_ if `$x` is indirectly an instance of `$A`.
+
+### relationship
+
+`$r ($A1: $x1, ..., $An: $xn)` is _satisfied_ if `$r` is a relation where for each `$Ai: $xi`, `$xi` is a role-player
+in `$r` indirectly playing the role `$Ai`.
+
+Additionally, for all pairs `$Ai: $xi` and `$Aj: $xj` where `i != j`, `$xi` and `$xj` must be different, or directly
+playing different roles in the relationship `$r`.
+
+The [variable](./query.html#variable) representing the role can optionally be omitted. If it is omitted, then the first
+[variable](./query.html#variable) is implicitly bound to the label `role`.
+
+### has (data)
+
+`$x has <identifier> $y as $r` is _satisfied_ if `$x` is related to an attribute `$y` of type `<identifier>`, where
+`$r` represents the relationship between them.
+
+This is equivalent to the following:
+<!-- TODO: Describe without referring to relationships? -->
```

Comment:
In the one meeting we agreed on the narrative that links to attributes have the potential to become relationships. So we kind of hiding the fact that they are relationships.

Reply:
Given this is a reference doc, I'm not totally sure how to half-hide this information... maybe it can go in its own section somewhere.